release.js

```diff
@@ -0,0 +1,90 @@
+const ChildProcess = require('child_process');
+const Fs = require('fs');
+
+const exec = cmd => new Promise(
+  (resolve, reject) => {
+    ChildProcess.exec(cmd, (err, stdout, stderr) => {
+      if (err != null) {
+        reject(err);
+      } else {
+        resolve({ stdout, stderr });
+      }
+    });
+  }
+);
+
+const readFile = (fileName, encodingType) => new Promise (
+  (resolve, reject) => {
+    Fs.readFile(fileName, encodingType, (err, data) => {
+      if (err != null) { reject(err); } else { resolve(data); }
+    });
+  }
+);
+
+const writeFile = (fileName, data) => new Promise (
+  (resolve, reject) => {
+    Fs.writeFile(fileName, data, err => {
+      if (err != null) { reject(err); } else { resolve(); }
+    });
+  }
+);
+
+const opamFile = "opam";
+const metaFile = "pkg/META";
+const packageJSONFile = "package.json";
+
+const writeFiles = async version => {
+  const opam = `
+opam-version: "1.2"
+name: "immutable"
+version: "${ version }"
+maintainer: "Dave Bordoley <bordoley>"
+authors: [ "Dave Bordoley <bordoley>" ]
+license: "BSD"
+homepage: "https://github.com/facebookincubator/immutable-re.git"
+dev-repo: "git://github.com/facebookincubator/immutable-re.git"
+bug-reports: "https://github.com/facebookincubator/immutable-re/issues"
+tags: [ "reason" "immutable" ]
+build: [
+  [make "build"]
+]
+depends: [
+  "topkg"       {build >=  "0.8.1" & < "0.9"}
+  "reason"      {build >=  "1.13.3"}
+]
+available: [ ocaml-version >= "4.02" & ocaml-version < "4.05" ]
+`;
+
+  const meta =  `
+# Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+
+version = "${ version }"
+description = "immutable: Immutable data structures for Reason"
+
+archive(byte) = "immutable.cma"
+archive(native) = "immutable.cmxa"
+`;
+
+  await writeFile (opamFile, opam);
+  await writeFile (metaFile, meta);
+};
+
+const main = async () => {
+  const packageJSON = await readFile (packageJSONFile, "UTF-8");
+  const version = JSON.parse(packageJSON).version;
+
+  await writeFiles (version);
+
```

Comment:
I found for myself that it was easier defining the version in the package.json itself and then generating the META and opam files. I don't think one is necessarily better than the other, but I'm more accustom to JS (npm) workflows.